lib/puppet/functions/azure_key_vault/lookup.rb

```diff
@@ -11,11 +11,11 @@ def lookup_key(secret_name, options, context)
     # This is a reserved key name in hiera
     return context.not_found if secret_name == 'lookup_options'
     return context.cached_value(secret_name) if context.cache_has_key(secret_name)
-    access_token = if context.cache_has_key('access_token')
-                     context.cached_value('access_token')
-                   else
-                     TragicCode::Azure.get_access_token(options['metadata_api_version'])
-                   end
+    access_token = context.cached_value('access_token')
+    if access_token.nil?
```

Comment:
wow, i cannot believe i actually never cached the value!!!

Can you write a unit test (https://github.com/TraGicCode/tragiccode-azure_key_vault/blob/master/spec/functions/azure_key_vault_lookup_spec.rb) to assert that the value is stored in the cache when it's not returned from the cache?